CRITICAL - 2023-01-07 14:12:52 --> The framework needs the following extension(s) installed and loaded: intl.
in SYSTEMPATH\CodeIgniter.php on line 224.
 1 SYSTEMPATH\CodeIgniter.php(224): CodeIgniter\Exceptions\FrameworkException::forMissingExtension('intl')
 2 SYSTEMPATH\CodeIgniter.php(182): CodeIgniter\CodeIgniter->resolvePlatformExtensions()
 3 ROOTPATH\spark(83): CodeIgniter\CodeIgniter->initialize()
CRITICAL - 2023-01-07 07:24:31 --> Undefined constant "CodeIgniter\Debug\VENDORPATH"
in SYSTEMPATH\Debug\Exceptions.php on line 177.
 1 SYSTEMPATH\Debug\Exceptions.php(157): CodeIgniter\Debug\Exceptions->isFakerDeprecationError(8192, 'number_format(): Passing null to parameter #1 ($num) of type float is deprecated', 'APPPATH\\Views\\Homepage\\inc\\cart\\itemcart.php', 48)
 2 [internal function]: CodeIgniter\Debug\Exceptions->errorHandler(8192, 'number_format(): Passing null to parameter #1 ($num) of type float is deprecated', 'APPPATH\\Views\\Homepage\\inc\\cart\\itemcart.php', 48)
 3 APPPATH\Views\Homepage\inc\cart\itemcart.php(48): number_format(null, 2)
 4 SYSTEMPATH\View\View.php(213): include('APPPATH\\Views\\Homepage\\inc\\cart\\itemcart.php')
 5 SYSTEMPATH\View\View.php(216): CodeIgniter\View\View->CodeIgniter\View\{closure}()
 6 SYSTEMPATH\View\View.php(431): CodeIgniter\View\View->render('Homepage/inc/cart/itemcart', null, true)
 7 APPPATH\Views\Homepage\cart.php(4): CodeIgniter\View\View->include('Homepage/inc/cart/itemcart')
 8 SYSTEMPATH\View\View.php(213): include('APPPATH\\Views\\Homepage\\cart.php')
 9 SYSTEMPATH\View\View.php(216): CodeIgniter\View\View->CodeIgniter\View\{closure}()
10 SYSTEMPATH\Common.php(1134): CodeIgniter\View\View->render('Homepage/cart', [], true)
11 APPPATH\Controllers\home.php(57): view('Homepage/cart', [...])
12 SYSTEMPATH\CodeIgniter.php(927): App\Controllers\home->cart()
13 SYSTEMPATH\CodeIgniter.php(482): CodeIgniter\CodeIgniter->runController(Object(App\Controllers\home))
14 SYSTEMPATH\CodeIgniter.php(351): CodeIgniter\CodeIgniter->handleRequest(null, Object(Config\Cache), false)
15 FCPATH\index.php(67): CodeIgniter\CodeIgniter->run()
16 SYSTEMPATH\Commands\Server\rewrite.php(46): require_once('FCPATH\\index.php')
CRITICAL - 2023-01-07 07:25:01 --> Trying to access array offset on value of type null
in APPPATH\Controllers\home.php on line 78.
 1 APPPATH\Controllers\home.php(78): CodeIgniter\Debug\Exceptions->errorHandler(2, 'Trying to access array offset on value of type null', 'APPPATH\\Controllers\\home.php', 78)
 2 SYSTEMPATH\CodeIgniter.php(927): App\Controllers\home->single_product('FCPATH\\assets', 'img', 'logo.png')
 3 SYSTEMPATH\CodeIgniter.php(482): CodeIgniter\CodeIgniter->runController(Object(App\Controllers\home))
 4 SYSTEMPATH\CodeIgniter.php(351): CodeIgniter\CodeIgniter->handleRequest(null, Object(Config\Cache), false)
 5 FCPATH\index.php(67): CodeIgniter\CodeIgniter->run()
 6 SYSTEMPATH\Commands\Server\rewrite.php(46): require_once('FCPATH\\index.php')
CRITICAL - 2023-01-07 07:25:01 --> Trying to access array offset on value of type null
in APPPATH\Controllers\home.php on line 78.
 1 APPPATH\Controllers\home.php(78): CodeIgniter\Debug\Exceptions->errorHandler(2, 'Trying to access array offset on value of type null', 'APPPATH\\Controllers\\home.php', 78)
 2 SYSTEMPATH\CodeIgniter.php(927): App\Controllers\home->single_product('FCPATH\\assets', 'img', 'products', 'product-img-1.jpg')
 3 SYSTEMPATH\CodeIgniter.php(482): CodeIgniter\CodeIgniter->runController(Object(App\Controllers\home))
 4 SYSTEMPATH\CodeIgniter.php(351): CodeIgniter\CodeIgniter->handleRequest(null, Object(Config\Cache), false)
 5 FCPATH\index.php(67): CodeIgniter\CodeIgniter->run()
 6 SYSTEMPATH\Commands\Server\rewrite.php(46): require_once('FCPATH\\index.php')
CRITICAL - 2023-01-07 07:25:01 --> Trying to access array offset on value of type null
in APPPATH\Controllers\home.php on line 78.
 1 APPPATH\Controllers\home.php(78): CodeIgniter\Debug\Exceptions->errorHandler(2, 'Trying to access array offset on value of type null', 'APPPATH\\Controllers\\home.php', 78)
 2 SYSTEMPATH\CodeIgniter.php(927): App\Controllers\home->single_product('FCPATH\\assets', 'img', 'products', 'product-img-2.jpg')
 3 SYSTEMPATH\CodeIgniter.php(482): CodeIgniter\CodeIgniter->runController(Object(App\Controllers\home))
 4 SYSTEMPATH\CodeIgniter.php(351): CodeIgniter\CodeIgniter->handleRequest(null, Object(Config\Cache), false)
 5 FCPATH\index.php(67): CodeIgniter\CodeIgniter->run()
 6 SYSTEMPATH\Commands\Server\rewrite.php(46): require_once('FCPATH\\index.php')
CRITICAL - 2023-01-07 07:25:02 --> Trying to access array offset on value of type null
in APPPATH\Controllers\home.php on line 78.
 1 APPPATH\Controllers\home.php(78): CodeIgniter\Debug\Exceptions->errorHandler(2, 'Trying to access array offset on value of type null', 'APPPATH\\Controllers\\home.php', 78)
 2 SYSTEMPATH\CodeIgniter.php(927): App\Controllers\home->single_product('FCPATH\\assets', 'img', 'products', 'product-img-3.jpg')
 3 SYSTEMPATH\CodeIgniter.php(482): CodeIgniter\CodeIgniter->runController(Object(App\Controllers\home))
 4 SYSTEMPATH\CodeIgniter.php(351): CodeIgniter\CodeIgniter->handleRequest(null, Object(Config\Cache), false)
 5 FCPATH\index.php(67): CodeIgniter\CodeIgniter->run()
 6 SYSTEMPATH\Commands\Server\rewrite.php(46): require_once('FCPATH\\index.php')
CRITICAL - 2023-01-07 07:25:02 --> Trying to access array offset on value of type null
in APPPATH\Controllers\home.php on line 78.
 1 APPPATH\Controllers\home.php(78): CodeIgniter\Debug\Exceptions->errorHandler(2, 'Trying to access array offset on value of type null', 'APPPATH\\Controllers\\home.php', 78)
 2 SYSTEMPATH\CodeIgniter.php(927): App\Controllers\home->single_product('FCPATH\\assets', 'img', 'favicon.png')
 3 SYSTEMPATH\CodeIgniter.php(482): CodeIgniter\CodeIgniter->runController(Object(App\Controllers\home))
 4 SYSTEMPATH\CodeIgniter.php(351): CodeIgniter\CodeIgniter->handleRequest(null, Object(Config\Cache), false)
 5 FCPATH\index.php(67): CodeIgniter\CodeIgniter->run()
 6 SYSTEMPATH\Commands\Server\rewrite.php(46): require_once('FCPATH\\index.php')
CRITICAL - 2023-01-07 10:25:56 --> Undefined constant "CodeIgniter\Debug\VENDORPATH"
in SYSTEMPATH\Debug\Exceptions.php on line 177.
 1 SYSTEMPATH\Debug\Exceptions.php(157): CodeIgniter\Debug\Exceptions->isFakerDeprecationError(8192, 'strpos(): Passing null to parameter #1 ($haystack) of type string is deprecated', 'SYSTEMPATH\\Email\\Email.php', 736)
 2 [internal function]: CodeIgniter\Debug\Exceptions->errorHandler(8192, 'strpos(): Passing null to parameter #1 ($haystack) of type string is deprecated', 'SYSTEMPATH\\Email\\Email.php', 736)
 3 SYSTEMPATH\Email\Email.php(736): strpos(null, ',')
 4 SYSTEMPATH\Email\Email.php(549): CodeIgniter\Email\Email->stringToArray(null)
 5 APPPATH\Controllers\Auth.php(168): CodeIgniter\Email\Email->setTo(null)
 6 SYSTEMPATH\CodeIgniter.php(927): App\Controllers\Auth->fpass()
 7 SYSTEMPATH\CodeIgniter.php(482): CodeIgniter\CodeIgniter->runController(Object(App\Controllers\Auth))
 8 SYSTEMPATH\CodeIgniter.php(351): CodeIgniter\CodeIgniter->handleRequest(null, Object(Config\Cache), false)
 9 FCPATH\index.php(67): CodeIgniter\CodeIgniter->run()
10 SYSTEMPATH\Commands\Server\rewrite.php(46): require_once('FCPATH\\index.php')
CRITICAL - 2023-01-07 10:27:09 --> Undefined constant "CodeIgniter\Debug\VENDORPATH"
in SYSTEMPATH\Debug\Exceptions.php on line 177.
 1 SYSTEMPATH\Debug\Exceptions.php(157): CodeIgniter\Debug\Exceptions->isFakerDeprecationError(8192, 'strpos(): Passing null to parameter #1 ($haystack) of type string is deprecated', 'SYSTEMPATH\\Email\\Email.php', 736)
 2 [internal function]: CodeIgniter\Debug\Exceptions->errorHandler(8192, 'strpos(): Passing null to parameter #1 ($haystack) of type string is deprecated', 'SYSTEMPATH\\Email\\Email.php', 736)
 3 SYSTEMPATH\Email\Email.php(736): strpos(null, ',')
 4 SYSTEMPATH\Email\Email.php(549): CodeIgniter\Email\Email->stringToArray(null)
 5 APPPATH\Controllers\Auth.php(168): CodeIgniter\Email\Email->setTo(null)
 6 SYSTEMPATH\CodeIgniter.php(927): App\Controllers\Auth->fpass()
 7 SYSTEMPATH\CodeIgniter.php(482): CodeIgniter\CodeIgniter->runController(Object(App\Controllers\Auth))
 8 SYSTEMPATH\CodeIgniter.php(351): CodeIgniter\CodeIgniter->handleRequest(null, Object(Config\Cache), false)
 9 FCPATH\index.php(67): CodeIgniter\CodeIgniter->run()
10 SYSTEMPATH\Commands\Server\rewrite.php(46): require_once('FCPATH\\index.php')
ERROR - 2023-01-07 10:29:36 --> Email: sendWithMail throwed ErrorException: mail(): Failed to connect to mailserver at &quot;localhost&quot; port 25, verify your &quot;SMTP&quot; and &quot;smtp_port&quot; setting in php.ini or use ini_set() in C:\xampp\htdocs\TeaTime\system\Email\Email.php:1730
Stack trace:
#0 [internal function]: CodeIgniter\Debug\Exceptions->errorHandler(2, 'mail(): Failed ...', 'C:\\xampp\\htdocs...', 1730)
#1 C:\xampp\htdocs\TeaTime\system\Email\Email.php(1730): mail('jamesducusin090...', '=?UTF-8?Q?<stro...', 'The OTP for you...', 'Date: Sat, 7 Ja...', '-f johnrexmalik...')
#2 C:\xampp\htdocs\TeaTime\system\Email\Email.php(1662): CodeIgniter\Email\Email->sendWithMail()
#3 C:\xampp\htdocs\TeaTime\system\Email\Email.php(1560): CodeIgniter\Email\Email->spoolEmail()
#4 C:\xampp\htdocs\TeaTime\app\Controllers\Auth.php(158): CodeIgniter\Email\Email->send()
#5 C:\xampp\htdocs\TeaTime\system\CodeIgniter.php(927): App\Controllers\Auth->fpass()
#6 C:\xampp\htdocs\TeaTime\system\CodeIgniter.php(482): CodeIgniter\CodeIgniter->runController(Object(App\Controllers\Auth))
#7 C:\xampp\htdocs\TeaTime\system\CodeIgniter.php(351): CodeIgniter\CodeIgniter->handleRequest(NULL, Object(Config\Cache), false)
#8 C:\xampp\htdocs\TeaTime\public\index.php(67): CodeIgniter\CodeIgniter->run()
#9 C:\xampp\htdocs\TeaTime\system\Commands\Server\rewrite.php(46): require_once('C:\\xampp\\htdocs...')
#10 {main}
ERROR - 2023-01-07 10:29:36 --> Email: Unable to send email using PHP mail(). Your server might not be configured to send mail using this method.
ERROR - 2023-01-07 10:29:36 --> 
ERROR - 2023-01-07 10:40:40 --> Email: sendWithMail throwed ErrorException: mail(): Failed to connect to mailserver at &quot;localhost&quot; port 25, verify your &quot;SMTP&quot; and &quot;smtp_port&quot; setting in php.ini or use ini_set() in C:\xampp\htdocs\TeaTime\system\Email\Email.php:1730
Stack trace:
#0 [internal function]: CodeIgniter\Debug\Exceptions->errorHandler(2, 'mail(): Failed ...', 'C:\\xampp\\htdocs...', 1730)
#1 C:\xampp\htdocs\TeaTime\system\Email\Email.php(1730): mail('jamesducusin090...', '=?UTF-8?Q?<stro...', 'The OTP for you...', 'Date: Sat, 7 Ja...', '-f johnrexmalik...')
#2 C:\xampp\htdocs\TeaTime\system\Email\Email.php(1662): CodeIgniter\Email\Email->sendWithMail()
#3 C:\xampp\htdocs\TeaTime\system\Email\Email.php(1560): CodeIgniter\Email\Email->spoolEmail()
#4 C:\xampp\htdocs\TeaTime\app\Controllers\Auth.php(158): CodeIgniter\Email\Email->send()
#5 C:\xampp\htdocs\TeaTime\system\CodeIgniter.php(927): App\Controllers\Auth->fpass()
#6 C:\xampp\htdocs\TeaTime\system\CodeIgniter.php(482): CodeIgniter\CodeIgniter->runController(Object(App\Controllers\Auth))
#7 C:\xampp\htdocs\TeaTime\system\CodeIgniter.php(351): CodeIgniter\CodeIgniter->handleRequest(NULL, Object(Config\Cache), false)
#8 C:\xampp\htdocs\TeaTime\public\index.php(67): CodeIgniter\CodeIgniter->run()
#9 C:\xampp\htdocs\TeaTime\system\Commands\Server\rewrite.php(46): require_once('C:\\xampp\\htdocs...')
#10 {main}
ERROR - 2023-01-07 10:40:40 --> Email: Unable to send email using PHP mail(). Your server might not be configured to send mail using this method.
ERROR - 2023-01-07 10:40:40 --> 
ERROR - 2023-01-07 10:43:06 --> Email: sendWithMail throwed ErrorException: mail(): Failed to connect to mailserver at &quot;localhost&quot; port 25, verify your &quot;SMTP&quot; and &quot;smtp_port&quot; setting in php.ini or use ini_set() in C:\xampp\htdocs\TeaTime\system\Email\Email.php:1730
Stack trace:
#0 [internal function]: CodeIgniter\Debug\Exceptions->errorHandler(2, 'mail(): Failed ...', 'C:\\xampp\\htdocs...', 1730)
#1 C:\xampp\htdocs\TeaTime\system\Email\Email.php(1730): mail('jamesducusin090...', '=?UTF-8?Q?<stro...', 'The OTP for you...', 'Date: Sat, 7 Ja...', '-f johnrexmalik...')
#2 C:\xampp\htdocs\TeaTime\system\Email\Email.php(1662): CodeIgniter\Email\Email->sendWithMail()
#3 C:\xampp\htdocs\TeaTime\system\Email\Email.php(1560): CodeIgniter\Email\Email->spoolEmail()
#4 C:\xampp\htdocs\TeaTime\app\Controllers\Auth.php(158): CodeIgniter\Email\Email->send()
#5 C:\xampp\htdocs\TeaTime\system\CodeIgniter.php(927): App\Controllers\Auth->fpass()
#6 C:\xampp\htdocs\TeaTime\system\CodeIgniter.php(482): CodeIgniter\CodeIgniter->runController(Object(App\Controllers\Auth))
#7 C:\xampp\htdocs\TeaTime\system\CodeIgniter.php(351): CodeIgniter\CodeIgniter->handleRequest(NULL, Object(Config\Cache), false)
#8 C:\xampp\htdocs\TeaTime\public\index.php(67): CodeIgniter\CodeIgniter->run()
#9 C:\xampp\htdocs\TeaTime\system\Commands\Server\rewrite.php(46): require_once('C:\\xampp\\htdocs...')
#10 {main}
ERROR - 2023-01-07 10:43:06 --> Email: Unable to send email using PHP mail(). Your server might not be configured to send mail using this method.
ERROR - 2023-01-07 10:43:06 --> 
ERROR - 2023-01-07 10:43:49 --> Email: sendWithMail throwed ErrorException: mail(): Failed to connect to mailserver at &quot;localhost&quot; port 25, verify your &quot;SMTP&quot; and &quot;smtp_port&quot; setting in php.ini or use ini_set() in C:\xampp\htdocs\TeaTime\system\Email\Email.php:1730
Stack trace:
#0 [internal function]: CodeIgniter\Debug\Exceptions->errorHandler(2, 'mail(): Failed ...', 'C:\\xampp\\htdocs...', 1730)
#1 C:\xampp\htdocs\TeaTime\system\Email\Email.php(1730): mail('jamesducusin090...', '=?UTF-8?Q?<stro...', 'The OTP for you...', 'Date: Sat, 7 Ja...', '-f johnrexmalik...')
#2 C:\xampp\htdocs\TeaTime\system\Email\Email.php(1662): CodeIgniter\Email\Email->sendWithMail()
#3 C:\xampp\htdocs\TeaTime\system\Email\Email.php(1560): CodeIgniter\Email\Email->spoolEmail()
#4 C:\xampp\htdocs\TeaTime\app\Controllers\Auth.php(158): CodeIgniter\Email\Email->send()
#5 C:\xampp\htdocs\TeaTime\system\CodeIgniter.php(927): App\Controllers\Auth->fpass()
#6 C:\xampp\htdocs\TeaTime\system\CodeIgniter.php(482): CodeIgniter\CodeIgniter->runController(Object(App\Controllers\Auth))
#7 C:\xampp\htdocs\TeaTime\system\CodeIgniter.php(351): CodeIgniter\CodeIgniter->handleRequest(NULL, Object(Config\Cache), false)
#8 C:\xampp\htdocs\TeaTime\public\index.php(67): CodeIgniter\CodeIgniter->run()
#9 C:\xampp\htdocs\TeaTime\system\Commands\Server\rewrite.php(46): require_once('C:\\xampp\\htdocs...')
#10 {main}
ERROR - 2023-01-07 10:43:49 --> Email: Unable to send email using PHP mail(). Your server might not be configured to send mail using this method.
ERROR - 2023-01-07 10:43:49 --> 
ERROR - 2023-01-07 10:44:23 --> Email: sendWithMail throwed ErrorException: mail(): Failed to connect to mailserver at &quot;localhost&quot; port 25, verify your &quot;SMTP&quot; and &quot;smtp_port&quot; setting in php.ini or use ini_set() in C:\xampp\htdocs\TeaTime\system\Email\Email.php:1730
Stack trace:
#0 [internal function]: CodeIgniter\Debug\Exceptions->errorHandler(2, 'mail(): Failed ...', 'C:\\xampp\\htdocs...', 1730)
#1 C:\xampp\htdocs\TeaTime\system\Email\Email.php(1730): mail('jamesducusin090...', '=?UTF-8?Q?<stro...', 'The OTP for you...', 'Date: Sat, 7 Ja...', '-f johnrexmalik...')
#2 C:\xampp\htdocs\TeaTime\system\Email\Email.php(1662): CodeIgniter\Email\Email->sendWithMail()
#3 C:\xampp\htdocs\TeaTime\system\Email\Email.php(1560): CodeIgniter\Email\Email->spoolEmail()
#4 C:\xampp\htdocs\TeaTime\app\Controllers\Auth.php(158): CodeIgniter\Email\Email->send()
#5 C:\xampp\htdocs\TeaTime\system\CodeIgniter.php(927): App\Controllers\Auth->fpass()
#6 C:\xampp\htdocs\TeaTime\system\CodeIgniter.php(482): CodeIgniter\CodeIgniter->runController(Object(App\Controllers\Auth))
#7 C:\xampp\htdocs\TeaTime\system\CodeIgniter.php(351): CodeIgniter\CodeIgniter->handleRequest(NULL, Object(Config\Cache), false)
#8 C:\xampp\htdocs\TeaTime\public\index.php(67): CodeIgniter\CodeIgniter->run()
#9 C:\xampp\htdocs\TeaTime\system\Commands\Server\rewrite.php(46): require_once('C:\\xampp\\htdocs...')
#10 {main}
ERROR - 2023-01-07 10:44:23 --> Email: Unable to send email using PHP mail(). Your server might not be configured to send mail using this method.
ERROR - 2023-01-07 10:44:23 --> 
ERROR - 2023-01-07 10:46:48 --> Email: sendWithMail throwed ErrorException: mail(): Failed to connect to mailserver at &quot;localhost&quot; port 25, verify your &quot;SMTP&quot; and &quot;smtp_port&quot; setting in php.ini or use ini_set() in C:\xampp\htdocs\TeaTime\system\Email\Email.php:1730
Stack trace:
#0 [internal function]: CodeIgniter\Debug\Exceptions->errorHandler(2, 'mail(): Failed ...', 'C:\\xampp\\htdocs...', 1730)
#1 C:\xampp\htdocs\TeaTime\system\Email\Email.php(1730): mail('jamesducusin090...', '=?UTF-8?Q?<stro...', 'The OTP for you...', 'Date: Sat, 7 Ja...', '-f johnrexmalik...')
#2 C:\xampp\htdocs\TeaTime\system\Email\Email.php(1662): CodeIgniter\Email\Email->sendWithMail()
#3 C:\xampp\htdocs\TeaTime\system\Email\Email.php(1560): CodeIgniter\Email\Email->spoolEmail()
#4 C:\xampp\htdocs\TeaTime\app\Controllers\Auth.php(158): CodeIgniter\Email\Email->send()
#5 C:\xampp\htdocs\TeaTime\system\CodeIgniter.php(927): App\Controllers\Auth->fpass()
#6 C:\xampp\htdocs\TeaTime\system\CodeIgniter.php(482): CodeIgniter\CodeIgniter->runController(Object(App\Controllers\Auth))
#7 C:\xampp\htdocs\TeaTime\system\CodeIgniter.php(351): CodeIgniter\CodeIgniter->handleRequest(NULL, Object(Config\Cache), false)
#8 C:\xampp\htdocs\TeaTime\public\index.php(67): CodeIgniter\CodeIgniter->run()
#9 C:\xampp\htdocs\TeaTime\system\Commands\Server\rewrite.php(46): require_once('C:\\xampp\\htdocs...')
#10 {main}
ERROR - 2023-01-07 10:46:48 --> Email: Unable to send email using PHP mail(). Your server might not be configured to send mail using this method.
ERROR - 2023-01-07 10:46:48 --> 
ERROR - 2023-01-07 10:56:01 --> Email: sendWithMail throwed ErrorException: mail(): Failed to connect to mailserver at &quot;localhost&quot; port 25, verify your &quot;SMTP&quot; and &quot;smtp_port&quot; setting in php.ini or use ini_set() in C:\xampp\htdocs\TeaTime\system\Email\Email.php:1730
Stack trace:
#0 [internal function]: CodeIgniter\Debug\Exceptions->errorHandler(2, 'mail(): Failed ...', 'C:\\xampp\\htdocs...', 1730)
#1 C:\xampp\htdocs\TeaTime\system\Email\Email.php(1730): mail('jamesducusin090...', '=?UTF-8?Q?<stro...', 'The OTP for you...', 'Date: Sat, 7 Ja...', '-f johnrexmalik...')
#2 C:\xampp\htdocs\TeaTime\system\Email\Email.php(1662): CodeIgniter\Email\Email->sendWithMail()
#3 C:\xampp\htdocs\TeaTime\system\Email\Email.php(1560): CodeIgniter\Email\Email->spoolEmail()
#4 C:\xampp\htdocs\TeaTime\app\Controllers\Auth.php(158): CodeIgniter\Email\Email->send()
#5 C:\xampp\htdocs\TeaTime\system\CodeIgniter.php(927): App\Controllers\Auth->fpass()
#6 C:\xampp\htdocs\TeaTime\system\CodeIgniter.php(482): CodeIgniter\CodeIgniter->runController(Object(App\Controllers\Auth))
#7 C:\xampp\htdocs\TeaTime\system\CodeIgniter.php(351): CodeIgniter\CodeIgniter->handleRequest(NULL, Object(Config\Cache), false)
#8 C:\xampp\htdocs\TeaTime\public\index.php(67): CodeIgniter\CodeIgniter->run()
#9 C:\xampp\htdocs\TeaTime\system\Commands\Server\rewrite.php(46): require_once('C:\\xampp\\htdocs...')
#10 {main}
ERROR - 2023-01-07 10:56:01 --> Email: Unable to send email using PHP mail(). Your server might not be configured to send mail using this method.
ERROR - 2023-01-07 10:56:01 --> 
ERROR - 2023-01-07 18:57:16 --> Error connecting to the database: mysqli_sql_exception: No connection could be made because the target machine actively refused it in C:\xampp\htdocs\TeaTime\system\Database\MySQLi\Connection.php:170
Stack trace:
#0 C:\xampp\htdocs\TeaTime\system\Database\MySQLi\Connection.php(170): mysqli->real_connect('localhost', 'root', Object(SensitiveParameterValue), 'project_ci4', 3306, '', 0)
#1 C:\xampp\htdocs\TeaTime\system\Database\BaseConnection.php(382): CodeIgniter\Database\MySQLi\Connection->connect(false)
#2 C:\xampp\htdocs\TeaTime\system\Database\BaseConnection.php(561): CodeIgniter\Database\BaseConnection->initialize()
#3 C:\xampp\htdocs\TeaTime\system\Database\BaseBuilder.php(1585): CodeIgniter\Database\BaseConnection->query('SELECT 1\nFROM `...', Array, false)
#4 C:\xampp\htdocs\TeaTime\system\Validation\Rules.php(104): CodeIgniter\Database\BaseBuilder->get()
#5 C:\xampp\htdocs\TeaTime\system\Validation\Validation.php(315): CodeIgniter\Validation\Rules->is_not_unique('johnrexgerard03...', 'email', Array, NULL, 'email')
#6 C:\xampp\htdocs\TeaTime\system\Validation\Validation.php(175): CodeIgniter\Validation\Validation->processRules('email', 'email', 'johnrexgerard03...', Array, Array)
#7 C:\xampp\htdocs\TeaTime\system\Controller.php(140): CodeIgniter\Validation\Validation->run()
#8 C:\xampp\htdocs\TeaTime\app\Controllers\Auth.php(163): CodeIgniter\Controller->validate(Array)
#9 C:\xampp\htdocs\TeaTime\system\CodeIgniter.php(927): App\Controllers\Auth->fpass()
#10 C:\xampp\htdocs\TeaTime\system\CodeIgniter.php(482): CodeIgniter\CodeIgniter->runController(Object(App\Controllers\Auth))
#11 C:\xampp\htdocs\TeaTime\system\CodeIgniter.php(351): CodeIgniter\CodeIgniter->handleRequest(NULL, Object(Config\Cache), false)
#12 C:\xampp\htdocs\TeaTime\public\index.php(67): CodeIgniter\CodeIgniter->run()
#13 C:\xampp\htdocs\TeaTime\system\Commands\Server\rewrite.php(46): require_once('C:\\xampp\\htdocs...')
#14 {main}

Next CodeIgniter\Database\Exceptions\DatabaseException: No connection could be made because the target machine actively refused it in C:\xampp\htdocs\TeaTime\system\Database\MySQLi\Connection.php:215
Stack trace:
#0 C:\xampp\htdocs\TeaTime\system\Database\BaseConnection.php(382): CodeIgniter\Database\MySQLi\Connection->connect(false)
#1 C:\xampp\htdocs\TeaTime\system\Database\BaseConnection.php(561): CodeIgniter\Database\BaseConnection->initialize()
#2 C:\xampp\htdocs\TeaTime\system\Database\BaseBuilder.php(1585): CodeIgniter\Database\BaseConnection->query('SELECT 1\nFROM `...', Array, false)
#3 C:\xampp\htdocs\TeaTime\system\Validation\Rules.php(104): CodeIgniter\Database\BaseBuilder->get()
#4 C:\xampp\htdocs\TeaTime\system\Validation\Validation.php(315): CodeIgniter\Validation\Rules->is_not_unique('johnrexgerard03...', 'email', Array, NULL, 'email')
#5 C:\xampp\htdocs\TeaTime\system\Validation\Validation.php(175): CodeIgniter\Validation\Validation->processRules('email', 'email', 'johnrexgerard03...', Array, Array)
#6 C:\xampp\htdocs\TeaTime\system\Controller.php(140): CodeIgniter\Validation\Validation->run()
#7 C:\xampp\htdocs\TeaTime\app\Controllers\Auth.php(163): CodeIgniter\Controller->validate(Array)
#8 C:\xampp\htdocs\TeaTime\system\CodeIgniter.php(927): App\Controllers\Auth->fpass()
#9 C:\xampp\htdocs\TeaTime\system\CodeIgniter.php(482): CodeIgniter\CodeIgniter->runController(Object(App\Controllers\Auth))
#10 C:\xampp\htdocs\TeaTime\system\CodeIgniter.php(351): CodeIgniter\CodeIgniter->handleRequest(NULL, Object(Config\Cache), false)
#11 C:\xampp\htdocs\TeaTime\public\index.php(67): CodeIgniter\CodeIgniter->run()
#12 C:\xampp\htdocs\TeaTime\system\Commands\Server\rewrite.php(46): require_once('C:\\xampp\\htdocs...')
#13 {main}
CRITICAL - 2023-01-07 18:57:16 --> Unable to connect to the database.
Main connection [MySQLi]: No connection could be made because the target machine actively refused it
in SYSTEMPATH\Database\BaseConnection.php on line 418.
 1 SYSTEMPATH\Database\BaseConnection.php(561): CodeIgniter\Database\BaseConnection->initialize()
 2 SYSTEMPATH\Database\BaseBuilder.php(1585): CodeIgniter\Database\BaseConnection->query('SELECT 1
FROM `user`
WHERE `email` = :email:
 LIMIT 1', [...], false)
 3 SYSTEMPATH\Validation\Rules.php(104): CodeIgniter\Database\BaseBuilder->get()
 4 SYSTEMPATH\Validation\Validation.php(315): CodeIgniter\Validation\Rules->is_not_unique('johnrexgerard03@gmail.com', 'email', [...], null, 'email')
 5 SYSTEMPATH\Validation\Validation.php(175): CodeIgniter\Validation\Validation->processRules('email', 'email', 'johnrexgerard03@gmail.com', [...], [...])
 6 SYSTEMPATH\Controller.php(140): CodeIgniter\Validation\Validation->run()
 7 APPPATH\Controllers\Auth.php(163): CodeIgniter\Controller->validate([...])
 8 SYSTEMPATH\CodeIgniter.php(927): App\Controllers\Auth->fpass()
 9 SYSTEMPATH\CodeIgniter.php(482): CodeIgniter\CodeIgniter->runController(Object(App\Controllers\Auth))
10 SYSTEMPATH\CodeIgniter.php(351): CodeIgniter\CodeIgniter->handleRequest(null, Object(Config\Cache), false)
11 FCPATH\index.php(67): CodeIgniter\CodeIgniter->run()
12 SYSTEMPATH\Commands\Server\rewrite.php(46): require_once('FCPATH\\index.php')
CRITICAL - 2023-01-07 19:44:47 --> count(): Argument #1 ($value) must be of type Countable|array, null given
in APPPATH\Controllers\Auth.php on line 196.
 1 APPPATH\Controllers\Auth.php(196): count(null)
 2 SYSTEMPATH\CodeIgniter.php(927): App\Controllers\Auth->otp()
 3 SYSTEMPATH\CodeIgniter.php(482): CodeIgniter\CodeIgniter->runController(Object(App\Controllers\Auth))
 4 SYSTEMPATH\CodeIgniter.php(351): CodeIgniter\CodeIgniter->handleRequest(null, Object(Config\Cache), false)
 5 FCPATH\index.php(67): CodeIgniter\CodeIgniter->run()
 6 SYSTEMPATH\Commands\Server\rewrite.php(46): require_once('FCPATH\\index.php')
CRITICAL - 2023-01-07 20:03:54 --> foreach() argument must be of type array|object, null given
in APPPATH\Controllers\Auth.php on line 202.
 1 APPPATH\Controllers\Auth.php(202): CodeIgniter\Debug\Exceptions->errorHandler(2, 'foreach() argument must be of type array|object, null given', 'APPPATH\\Controllers\\Auth.php', 202)
 2 SYSTEMPATH\CodeIgniter.php(927): App\Controllers\Auth->otp()
 3 SYSTEMPATH\CodeIgniter.php(482): CodeIgniter\CodeIgniter->runController(Object(App\Controllers\Auth))
 4 SYSTEMPATH\CodeIgniter.php(351): CodeIgniter\CodeIgniter->handleRequest(null, Object(Config\Cache), false)
 5 FCPATH\index.php(67): CodeIgniter\CodeIgniter->run()
 6 SYSTEMPATH\Commands\Server\rewrite.php(46): require_once('FCPATH\\index.php')
CRITICAL - 2023-01-07 20:06:20 --> Unsupported operand types: array + string
in APPPATH\Controllers\Auth.php on line 208.
 1 SYSTEMPATH\CodeIgniter.php(927): App\Controllers\Auth->otp()
 2 SYSTEMPATH\CodeIgniter.php(482): CodeIgniter\CodeIgniter->runController(Object(App\Controllers\Auth))
 3 SYSTEMPATH\CodeIgniter.php(351): CodeIgniter\CodeIgniter->handleRequest(null, Object(Config\Cache), false)
 4 FCPATH\index.php(67): CodeIgniter\CodeIgniter->run()
 5 SYSTEMPATH\Commands\Server\rewrite.php(46): require_once('FCPATH\\index.php')
CRITICAL - 2023-01-07 20:06:57 --> Unsupported operand types: array + string
in APPPATH\Controllers\Auth.php on line 208.
 1 SYSTEMPATH\CodeIgniter.php(927): App\Controllers\Auth->otp()
 2 SYSTEMPATH\CodeIgniter.php(482): CodeIgniter\CodeIgniter->runController(Object(App\Controllers\Auth))
 3 SYSTEMPATH\CodeIgniter.php(351): CodeIgniter\CodeIgniter->handleRequest(null, Object(Config\Cache), false)
 4 FCPATH\index.php(67): CodeIgniter\CodeIgniter->run()
 5 SYSTEMPATH\Commands\Server\rewrite.php(46): require_once('FCPATH\\index.php')
CRITICAL - 2023-01-07 20:55:41 --> There is no data to update.
in SYSTEMPATH\BaseModel.php on line 1598.
 1 SYSTEMPATH\BaseModel.php(1598): CodeIgniter\Database\Exceptions\DataException::forEmptyDataset('update')
 2 SYSTEMPATH\BaseModel.php(853): CodeIgniter\BaseModel->transformDataToArray(null, 'update')
 3 SYSTEMPATH\Model.php(693): CodeIgniter\BaseModel->update([...], null)
 4 APPPATH\Controllers\Auth.php(264): CodeIgniter\Model->update([...])
 5 SYSTEMPATH\CodeIgniter.php(927): App\Controllers\Auth->reset()
 6 SYSTEMPATH\CodeIgniter.php(482): CodeIgniter\CodeIgniter->runController(Object(App\Controllers\Auth))
 7 SYSTEMPATH\CodeIgniter.php(351): CodeIgniter\CodeIgniter->handleRequest(null, Object(Config\Cache), false)
 8 FCPATH\index.php(67): CodeIgniter\CodeIgniter->run()
 9 SYSTEMPATH\Commands\Server\rewrite.php(46): require_once('FCPATH\\index.php')
CRITICAL - 2023-01-07 21:01:10 --> CodeIgniter\Validation\Rules::min_length(): Argument #2 ($val) must be of type string, null given, called in C:\xampp\htdocs\TeaTime\system\Validation\Validation.php on line 314
in SYSTEMPATH\Validation\Rules.php on line 186.
 1 SYSTEMPATH\Validation\Validation.php(314): CodeIgniter\Validation\Rules->min_length(null, null)
 2 SYSTEMPATH\Validation\Validation.php(175): CodeIgniter\Validation\Validation->processRules('otp[]', 'otp[]', null, [...], [...])
 3 SYSTEMPATH\Controller.php(140): CodeIgniter\Validation\Validation->run()
 4 APPPATH\Controllers\Auth.php(210): CodeIgniter\Controller->validate([...])
 5 SYSTEMPATH\CodeIgniter.php(927): App\Controllers\Auth->otp()
 6 SYSTEMPATH\CodeIgniter.php(482): CodeIgniter\CodeIgniter->runController(Object(App\Controllers\Auth))
 7 SYSTEMPATH\CodeIgniter.php(351): CodeIgniter\CodeIgniter->handleRequest(null, Object(Config\Cache), false)
 8 FCPATH\index.php(67): CodeIgniter\CodeIgniter->run()
 9 SYSTEMPATH\Commands\Server\rewrite.php(46): require_once('FCPATH\\index.php')
CRITICAL - 2023-01-07 21:12:53 --> There is no data to update.
in SYSTEMPATH\BaseModel.php on line 1598.
 1 SYSTEMPATH\BaseModel.php(1598): CodeIgniter\Database\Exceptions\DataException::forEmptyDataset('update')
 2 SYSTEMPATH\BaseModel.php(853): CodeIgniter\BaseModel->transformDataToArray(null, 'update')
 3 SYSTEMPATH\Model.php(693): CodeIgniter\BaseModel->update([...], null)
 4 APPPATH\Controllers\Auth.php(248): CodeIgniter\Model->update([...])
 5 SYSTEMPATH\CodeIgniter.php(927): App\Controllers\Auth->reset()
 6 SYSTEMPATH\CodeIgniter.php(482): CodeIgniter\CodeIgniter->runController(Object(App\Controllers\Auth))
 7 SYSTEMPATH\CodeIgniter.php(351): CodeIgniter\CodeIgniter->handleRequest(null, Object(Config\Cache), false)
 8 FCPATH\index.php(67): CodeIgniter\CodeIgniter->run()
 9 SYSTEMPATH\Commands\Server\rewrite.php(46): require_once('FCPATH\\index.php')
CRITICAL - 2023-01-07 21:46:49 --> Undefined constant "CodeIgniter\Debug\VENDORPATH"
in SYSTEMPATH\Debug\Exceptions.php on line 177.
 1 SYSTEMPATH\Debug\Exceptions.php(157): CodeIgniter\Debug\Exceptions->isFakerDeprecationError(8192, 'number_format(): Passing null to parameter #1 ($num) of type float is deprecated', 'APPPATH\\Views\\Homepage\\inc\\cart\\itemcart.php', 48)
 2 [internal function]: CodeIgniter\Debug\Exceptions->errorHandler(8192, 'number_format(): Passing null to parameter #1 ($num) of type float is deprecated', 'APPPATH\\Views\\Homepage\\inc\\cart\\itemcart.php', 48)
 3 APPPATH\Views\Homepage\inc\cart\itemcart.php(48): number_format(null, 2)
 4 SYSTEMPATH\View\View.php(213): include('APPPATH\\Views\\Homepage\\inc\\cart\\itemcart.php')
 5 SYSTEMPATH\View\View.php(216): CodeIgniter\View\View->CodeIgniter\View\{closure}()
 6 SYSTEMPATH\View\View.php(431): CodeIgniter\View\View->render('Homepage/inc/cart/itemcart', null, true)
 7 APPPATH\Views\Homepage\cart.php(4): CodeIgniter\View\View->include('Homepage/inc/cart/itemcart')
 8 SYSTEMPATH\View\View.php(213): include('APPPATH\\Views\\Homepage\\cart.php')
 9 SYSTEMPATH\View\View.php(216): CodeIgniter\View\View->CodeIgniter\View\{closure}()
10 SYSTEMPATH\Common.php(1134): CodeIgniter\View\View->render('Homepage/cart', [], true)
11 APPPATH\Controllers\home.php(57): view('Homepage/cart', [...])
12 SYSTEMPATH\CodeIgniter.php(927): App\Controllers\home->cart()
13 SYSTEMPATH\CodeIgniter.php(482): CodeIgniter\CodeIgniter->runController(Object(App\Controllers\home))
14 SYSTEMPATH\CodeIgniter.php(351): CodeIgniter\CodeIgniter->handleRequest(null, Object(Config\Cache), false)
15 FCPATH\index.php(67): CodeIgniter\CodeIgniter->run()
16 SYSTEMPATH\Commands\Server\rewrite.php(46): require_once('FCPATH\\index.php')
